homeassistant/components/gentex_homelink/config_flow.py

```diff
@@ -68,3 +75,34 @@ async def async_step_user(
             ),
             errors=errors,
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+                ),
+            )
+        return await self.async_step_user(user_input)
+
+    async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+        await self.async_set_unique_id(self.external_data[CONF_UNIQUE_ID])
+        entry_title = self.external_data[CONF_EMAIL] or self.flow_impl.name
+        if self.source == SOURCE_REAUTH:
+            self._abort_if_unique_id_mismatch()
+            return self.async_update_reload_and_abort(
+                self._get_reauth_entry(), data_updates=data, title=entry_title
+            )
+        self._abort_if_unique_id_configured()
```

Comment:
Why do we have `self._abort_if_unique_id_configured()` twice? I think I would recommend to move the logic of the other one here